backends/ebpf/codeGen.cpp

```diff
@@ -343,32 +342,34 @@ bool CodeGenInspector::preorder(const IR::Type_Enum *type) {
 
 void CodeGenInspector::emitAssignStatement(const IR::Type *ltype, const IR::Expression *lexpr,
                                            cstring lpath, const IR::Expression *rexpr) {
-    auto ebpfType = EBPFTypeFactory::instance->create(ltype);
-    bool memcpy = false;
-    EBPFScalarType *scalar = nullptr;
-    unsigned width = 0;
-    if (ebpfType->is<EBPFScalarType>()) {
-        scalar = ebpfType->to<EBPFScalarType>();
-        width = scalar->implementationWidthInBits();
-        memcpy = !EBPFScalarType::generatesScalar(width);
-    }
     builder->emitIndent();
-    if (memcpy) {
-        builder->append("__builtin_memcpy(&");
-        if (lexpr != nullptr) {
-            visit(lexpr);
-        } else {
-            builder->append(lpath);
-        }
-        builder->append(", &");
-        if (rexpr->is<IR::Constant>()) {
-            builder->appendFormat("(u8[%u])", scalar->bytesRequired());
-        }
-        visit(rexpr);
-        builder->appendFormat(", %d)", scalar->bytesRequired());
+    if (builder->target->name == "P4TC") {
```

Comment:
Really worried about all this TC specific code. Is there no way you can't instantiate a TCCodeGenInspector at this point?